src/components/registration/RegistrationForm.tsx

```diff
@@ -1,73 +1,77 @@
 import React, { useContext, useState } from "react";
 import { Button, Typography } from "@material-ui/core";
-import PropTypes from "prop-types";
 import { DynamicFieldsContext } from "../../context/DynamicFieldsContext";
 import FormFieldGroup from "./FormFieldGroup";
 import {
   DefaultFamilyFormFields,
   DefaultStudentFormFields,
 } from "../../constants/DefaultFields";
 import StudentRole from "../../constants/StudentRole";
+import {
+  FamilyRequest,
+  FamilyStudentRequest,
+  StudentRequest,
+} from "../../api/FamilyAPI";
+import { DefaultFormField } from "../../hooks/useFormFields";
+import DefaultFieldKey from "../../constants/DefaultFieldKey";
+
+const defaultFamilyRequestData: FamilyRequest = {
+  [DefaultFieldKey.ADDRESS]: "",
+  [DefaultFieldKey.CELL_NUMBER]: "",
+  [DefaultFieldKey.EMAIL]: "",
+  [DefaultFieldKey.HOME_NUMBER]: "",
+  [DefaultFieldKey.PREFERRED_CONTACT]: "",
+  [DefaultFieldKey.PREFERRED_NUMBER]: "",
+  [DefaultFieldKey.WORK_NUMBER]: "",
+};
+
+const defaultStudentRequestData: StudentRequest = {
+  [DefaultFieldKey.FIRST_NAME]: "",
+  [DefaultFieldKey.LAST_NAME]: "",
+  information: {},
+};
+
+type RegistrationFormProps = {
+  onSubmit: (
+    e: React.FormEvent<HTMLFormElement>,
+    data: FamilyStudentRequest
+  ) => void;
+};
 
-function RegistrationForm({ onSubmit }) {
+const RegistrationForm = ({ onSubmit }: RegistrationFormProps) => {
   const {
     childDynamicFields,
     guestDynamicFields,
     parentDynamicFields,
   } = useContext(DynamicFieldsContext);
 
-  const [familyData, setFamilyData] = useState({});
-  const [parentData, setParentData] = useState({ information: {} });
-  const [childData, setChildData] = useState({ information: {} });
-  const [guestData, setGuestData] = useState({ information: {} });
+  const [familyData, setFamilyData] = useState<FamilyRequest>({
+    ...defaultFamilyRequestData,
+  });
+  const [parentData, setParentData] = useState<StudentRequest>({
+    ...defaultStudentRequestData,
+  });
+  const [childData, setChildData] = useState({ ...defaultStudentRequestData });
+  const [guestData, setGuestData] = useState({ ...defaultStudentRequestData });
```

Comment:
should we add the `<StudentRequest>` type for the hooks here?

Reply:
ooh good call, just added